src/formatDate.js

```diff
@@ -50,7 +50,77 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dayParts = date.split(fromFormat[fromFormat.length - 1]);
+
+  let day = '';
+  let month = '';
+  let year = '';
+  const devider = toFormat[toFormat.length - 1];
+  const formatDay = 'DD';
+  const formatMonth = 'MM';
+  const formatYearFourChars = 'YYYY';
+  const formatYearTwoChars = 'YY';
+  const toTwentyFirstCentury = 2000;
+  const toTwentiethCentury = 1900;
+  const limitYearForCentury = 30;
+
+  for (let i = 0; i < dayParts.length; i++) {
+    if (fromFormat[i] === formatDay) {
+      day = dayParts[i];
+    }
+
+    if (fromFormat[i] === formatMonth) {
+      month = dayParts[i];
+    }
+
+    if (fromFormat[i] === formatYearFourChars
+      || fromFormat[i] === formatYearTwoChars) {
+      year = dayParts[i];
+    }
+  }
+
+  let formattedDate = '';
+
+  for (let i = 0; i < toFormat.length; i++) {
+    if (toFormat[i] === formatYearFourChars) {
+      if (year.length === 2) {
+        let yearToNum = parseInt(year, 10);
+
+        if (yearToNum < limitYearForCentury) {
+          yearToNum += toTwentyFirstCentury;
+        } else {
+          yearToNum += toTwentiethCentury;
+        }
+
+        year = yearToNum.toString();
+      } else if (toFormat[i] === formatYearTwoChars) {
+        if (year.length === 4) {
+          year = year.slice(2);
+        }
+      }
+    }
```

Comment:
If possible, avoid large nested `if`s, it is difficult to read and maintain in the future.
```suggestion
    if (toFormat[i] === formatYearFourChars && year.length === 2) {
      let yearToNum = parseInt(year, 10);
    
      if (yearToNum < limitYearForCentury) {
        yearToNum += toTwentyFirstCentury;
      } else {
        yearToNum += toTwentiethCentury;
      }
    
      year = yearToNum.toString();
    }
  
    if (toFormat[i] === formatYearTwoChars && year.length === 4) {
      year = year.slice(2);
    }
```